pkg/volume/flexvolume/util.go

```diff
@@ -104,7 +104,7 @@ func getReadOnly(spec *volume.Spec) (bool, error) {
 	}
 	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.FlexVolume != nil {
 		// ReadOnly is specified at the PV level
-		return spec.ReadOnly, nil
+		return spec.PersistentVolume.Spec.FlexVolume.ReadOnly, nil
```

Comment:
Can you please elaborate on why is this the proper fix? Also, to double check, what are other volumes doing?

Honest question, it would really help me to easily review this :)

Reply:
the same logic exists in CSI and RBD... 

RBD has this comment:

// rbd volumes used as a PersistentVolume gets the ReadOnly flag indirectly through
// the persistent-claim volume used to mount the PV

it's unclear (to me) what `spec.ReadOnly` represents and which is the correct value to use. @saad-ali?

Reply:
@rata @liggitt @saad-ali I have updated this PR and explanation, readOnly should respects both `spec.PersistentVolume.Spec.FlexVolume.ReadOnly` and `Pod.Spec.Volumes.PersistentVolumeClaim.ReadOnly`:
```
return spec.ReadOnly || spec.PersistentVolume.Spec.FlexVolume.ReadOnly
```

Reply:
It seems that some other volume plugins have this issue too, such as AWS and GCE. We should give a clear definition for PVC's `ReadOnly` attribute. I post some thoughts in according [issue comment](https://github.com/kubernetes/kubernetes/issues/61758#issuecomment-376506621).